pkg/gather/clusterconfig/clusterconfig.go

```diff
@@ -123,10 +128,10 @@ func (i *Gatherer) Gather(ctx context.Context, recorder record.Interface) error
 //
 // The GET REST query to URL /federate
 // Gathered metrics:
-//   ALERTS
 //   etcd_object_counts
 //   cluster_installer
 //   namespace CPU and memory usage
+//   followed by at most 1137 lines of ALERTS metric
```

Comment:
is there any behavior for these 1137 lines? I would like to see the most recent alerts, and why not use a number like 1000?

Reply:
another question, maybe for @iNecas, why not stop collecting alerts since we already have it on Thanos.

We will need to query Thanos anyway to get all the alerts because the archive does not ship all alerts, why bother sending it anyway if sending just in half, thinking of pipeline perspective, maybe for others this makes sense but for us, it does not, I would never rely on IO data if not complete.

Reply:
@alexandrevicenzi Please see the constant above. It's a bit of an odd number, but on average, 1137 alerts would add up to 500 KiB of data.

Reply:
I think it still makes sense to have them but one good point: can we indicate that the data are incomplete? We can then have a look on how % of clusters are hitting this situation and make a decision whether this is the way or not.

Especially for rules writing, depending on external service is always a bit problematic, as you know @alexandrevicenzi. Therefore it's good to understand what real impact we're talking about here. On side node, if a cluster is hitting 1000+ alerts, that's probably the first thing to fix before anything else, and out of those alerts, we can still ask ourselves, if we can help the customer fixing that.

Reply:
@iNecas One thing I have discussed with Martin was that we could add the total number of alerts somewhere. WDYT? I've noticed there's a `# Alerts\n` label/comment above the alerts. Would it be possible to add a number after that or would it break some king of a pre-defined format?

Reply:
@natiiix I just find it an odd number, something even would be more interesting, if I get 1000 or 1137 is does not make much of a difference.

@iNecas, what if the first 1k alerts are from Watchdog and later on there's an alert from a real problem, this is the problem that I see, truncating it is as bad as not having at all, at least in my perspective.

I won't block anything, it was just a question, but as I mentioned, in the pipeline perspective we won't use or trust this data much, we will still query Thanos for the complete list of alerts.

I would be more interesting to parse the alerts or to get a list of unique alerts, I don't think we really need to know all the times that one issue happened, the last one seen is probably good enough, as well skipping Watchdog related alerts would be something to save some space.

Reply:
@alexandrevicenzi I don't know what this data is used for, but a large quantity of them clearly indicates an issue as far as I see. Maybe the metrics are used by external rules? I don't know. Someone wanted it there and now it's there, but it's not acceptable to have 70 MB of metrics, no matter who wants them.

Reply:
> what if the first 1k alerts are from Watchdog

Do you have real examples for this? I've not seen more than one for a cluster, if there were 100s of that, I think that would be also thing to fix.

Different approach could be  `sum without` promql, to eliminate labels we don't care and have more concise view, but that would mean we would be putting additional load on in-custer prometheus (might not be an issue, but raising it here). Anyway, I don't want to over-complicate this particular PR, limiting the lines seems like good enough to do the initial mitigation (bonus points for the number of all alerts somewhere)

Reply:
@iNecas When Martin tried to filter them out using the query, he wasn't very successful apparently. I don't know the specifics, but he said it probably wouldn't be a good way of doing it.

Reply:
To put the fire off this is ok, but a better solution would be probably more interesting in the long term, as I said, it was questions regarding the usefulness of the data in the pipeline, this is my concern.

Reply:
Adding here just to complete the context: https://github.com/prometheus/client_python/blob/3ab0374d1548769748932b9316d86b088dc7712e/prometheus_client/parser.py#L177 <- this is the parser @alexandrevicenzi meant that relies on specific form of the comment, so we should keep that untouched

Reply:
As for detecting the limit hit, for now, we could probably just guess from the number of lines: not great not terrible.

@natiiix @alexandrevicenzi one thing orthogonal to this, but that would help going forward would be putting more metadata about different collectors execution: I can think of execution time, raw files size, any errors etc., that we could use to store metadata about the collected data, and in this case could capture the fact that the data are incomplete + how many more alerts are there. Would you be willing to put some task in jira about this and see if that could be something to prioritize among other things on the teams plate?

Reply:
@iNecas Okay, that makes sense. I will create a task for it.